mcs/tools/mono-api-html/ApiDiff.cs

```diff
@@ -218,4 +172,99 @@ public static int Main (string[] args)
 			return 0;
 		}
 	}
+#endif
+
+	public enum ApiDiffFormatter {
+		Html,
+		Markdown,
+	}
+
+	public class ApiDiffFormattedConfig {
```

Comment:
The `ApiDiffFormattedConfig` type is new and different from `State` in that `State` also contains temporary data during the actual processing.

I _can_ merge the types and make things internal and all that, but this way there is a clear separation - and this config is never modified by the library.